channeldb/payments.go

```diff
@@ -596,7 +596,10 @@ func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
 		// Counting the total number of payments is expensive, since we
 		// literally have to traverse the cursor linearly, which can
 		// take quite a while. So it's an optional query parameter.
-		if query.CountTotal {
+		if query.CountTotal && query.CreationDateStart == 0 &&
+			query.CreationDateEnd == 0 {
+			// It will only take effect if CreationDateStart and
+			// CreationDateEnd are not specified.
 			var (
```

Comment:
I dont think this is the behaviour we want. I think that we still want to count how many payments there were _but_ we just want to count the ones that happened between the given start and end times.

Reply:
thanks for the feedback @ellemouton 

the original bug report by @andyschroder specifies that 
```
$ lncli listpayments --creation_date_start 1650000000 --count_total_payments --creation_date_end 1650000001|grep total_num_payments
    "total_num_payments": "0"
$
```
it seems like the original issue had a different interpretation on the expected behaviour

Reply:
yes, that time difference was specified to assume there were no payments between that short period of time.

Reply:
> I dont think this is the behaviour we want. I think that we still want to count how many payments there were _but_ we just want to count the ones that happened between the given start and end times.

yes, this was the original purpose of the bug

Reply:
> thanks for the feedback @ellemouton
> 
> the original bug report by @AndySchroder specifies that
> 
> ```
> $ lncli listpayments --creation_date_start 1650000000 --count_total_payments --creation_date_end 1650000001|grep total_num_payments
>     "total_num_payments": "0"
> $
> ```
> 
> it seems like the original issue had a different interpretation on the expected behaviour

no, please understand the interpretation that @ellemouton is providing.